src/Data/Maybe.agda

```diff
@@ -56,6 +57,29 @@ monadPlus {f} = record
   nothing ∣ y = y
   just x  ∣ y = just x
 
+
+------------------------------------------------------------------------
+-- Other associated operations
+
+fromMaybe : {ℓ : Level} {A : Set ℓ} (default : A) (try : Maybe A) → A
+fromMaybe def (just x) = x
+fromMaybe def nothing  = def
+
+listToMaybe : {ℓ : Level} {A : Set ℓ} → List A → Maybe A
+listToMaybe []       = nothing
+listToMaybe (x ∷ xs) = just x
+
+maybeToList : {ℓ : Level} {A : Set ℓ} → Maybe A → List A
+maybeToList (just x) = [ x ]
+maybeToList nothing  = []
+
+catMaybes : {ℓ : Level} {A : Set ℓ} → List (Maybe A) → List A
+catMaybes []             = []
+catMaybes (just x ∷ xs)  = x ∷ catMaybes xs
+catMaybes (nothing ∷ xs) = catMaybes xs
+
```

Comment:
Note that this could be implemented as `mapMaybe id`

Reply:
Yes. `Maybe` is viewed as more fundamental (whatever that means!) than `List` and so any function involving both of them should go in `Data.List.Base` rather than `Data.Maybe.Base`.

Reply:
That's because Pointed is more fundamental than Monoid -- but that's a long adjoint story best left for another time.